book/04-latex/latex-figure-positioning.Rmd

```diff
@@ -1,22 +1,96 @@
 
-<!---
+- `h`:	Place the float here, i.e., approximately at the same point it occurs in the source text (however, not exactly at the spot)
+- `t`:	Position at the top of the page.
+- `b`:	Position at the bottom of the page.
+- `p`:	Put on a special page for floats only.
+- `!`:	Override internal parameters LaTeX uses for determining "good" float positions.
+- `H`: Places the float at precisely the location in the LaTeX code. Requires the float package (`\usepackage{float}`)
 
-### Controlling Float
+The default behaviour for LaTeX in R Markdown is `tbp`, that is it will try positioning the figure at the top of the page, then then bottom, and if not it will put them on a separate page for floats.
 
+### Preventing figures floating
 
+Many users will initially want to prevent figures from floating in their document, replicating the behaviour of a traditional word processor. To do this, we must firstly load the LaTeX package `float`. This can be done by including the following line in the YAML:
 
-### Forcing the position
+```
+output: 
+pdf_document:
+extra_dependencies: ["float"]
+```
 
+We can use the `fig.pos` to control the float behaviour, and the use of `!H` will prevent any floating within the document. We can set the default behaviour for the document so that all chunks have this setting by including the following line at the start of your R Markdown document:
 
+```{r, echo=TRUE, eval=FALSE}
+knitr::opts_chunk$set(fig.pos = "!H")
+```
 
+Please note, that we would strongly advise that readers consider the behaviour of float in their document. This solution was included within the book by popular demand^[the related[StackOverflow question](https://stackoverflow.com/q/16626462/7347699) has been viewed over 40000 times] but there are some serious sides effects overriding the default settings. Primarily, you may end 
+
+### Force floats forwards
+<!--- https://tex.stackexchange.com/questions/15706/force-floats-to-be-typeset-after-their-occurrence-in-the-source-text --->
+
+An alternative to forcing all floats to be held is to force floats forwards in the text. This can remove a common issue, where a figure is shown at the top of the page on which it is created. This can break the flow of a report. We can force it so that the figure always appears after the text by using the **flafter** LaTeX package as follows:
+
+```
+output: 
+pdf_document:
+extra_dependencies: ["flafter"]
+```
+
+### Adjust LaTeX placement rules
+
+LaTeX’s own float placement parameters could be preventing placements that seem entirely “reasonable” to you — they’re notoriously rather conservative. These defaults are displayed in Table \@ref(tab:latexFloatDefault).
+
+<!--- This table may work better as a diagram? --->
+
+```{r latexFloatDefault}
```

Comment:
I think we need to force `echo = FALSE` as we don't want the code 
```suggestion
```{r latexFloatDefault, echo = FALSE}
```